### PR TITLE
OOP school library: add basic UI

### PR DESCRIPTION
**Pull request by Jrmcuervo**
### Build interactive console apps and run a program using the command line.

In this PR I;

- Created an app.rb file that serves as the entry point for my console app, It has methods that do the following:
  1. List all books.
  2. List all people.
  3. Create a person (teacher or student, not a plain Person).
  4. Create a book.
  5. Create a rental.
  6. List all rentals for a given person id.

- Created a main.rb file with a method called main that is called at the end of its file. This method does the following:
  1. Present the user with a list of options to perform.
  2. Lets users choose an option.
  3. If needed, ask for parameters for the option.
  4. Have a way to quit the app.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    person_to_rent = @people.find { |p| p.id == person_id }
+
+    return puts "No book with title: #{book_title}" if book_to_rent.nil?
+    return puts "No person with ID: #{person_id}" if person_to_rent.nil?
+
+    rental = Rental.new(date, book_to_rent, person_to_rent)
+    @rentals << rental
+    puts 'Rental created successfully'
+  end
+
```
Comment (cynthiainga) [inline app.rb]: - [ ] Good job so far but to create a rental we need to select existing books and persons listed by id as shown in the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ), please kindly fix that 🙏🏾 

How we can create a rental in your app 👇🏾 

https://github.com/Jrmcuervo/OPP-school-library/assets/74501523/1ec6e0c7-9c59-4e5e-b33d-33530dd318ae

- [ ] Also make sure the list of rentals is displayed as in the given video 🙃 😉

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_all_people
+    @people.each do |person|
+      puts "ID: #{person.id}, Name: #{person.name}, Age: #{person.age}"
+    end
+  end
```
Comment (cynthiainga) [inline app.rb]: - [ ] Good job so far but when listing people should display differently, it should be precise if the person is a student or a teacher, kindly fix that 🙏🏾 

Yours

Screenshot 2023-07-27 at 21 41 54

In the original video

Screenshot 2023-07-27 at 21 42 10

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @Jrmcuervo, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] No linters errors ✔️ 
- [ ] Descriptive PR ✔️ 
- [ ] Correct Gitflow ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by IngDavidTM — APPROVED** [review]
Hi @Jrmcuervo 🙋‍♂️, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request follows the GitFlow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**